selinux: disable by default

be6ec88f41ba94746f830ba38cc4d08dd5ddbb08 "selinux: Remove
SECURITY_SELINUX_BOOTPARAM_VALUE" has effectively enabled selinux
unconditionally by default, which has unintended consequences even if
there are no selinux policies provided by userspace (e.g. preventing
rsync from successfully copying all files of the currently booted
rootfs, due to lsetxattr not being allowed to set the security.selinux
label).

selinux can be re-enabled by explicitly setting selinux=1 on the
CMDLINE or (partially) by writing 0 to /sys/fs/selinux/disable.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -124,7 +124,7 @@ __setup("enforcing=", enforcing_setup);
 #define selinux_enforcing_boot 1
 #endif
 
-int selinux_enabled_boot __initdata = 1;
+int selinux_enabled_boot __initdata = 0;
 #ifdef CONFIG_SECURITY_SELINUX_BOOTPARAM
 static int __init selinux_enabled_setup(char *str)
 {
